File: scripts/run_migrations.py
#!/usr/bin/env python3

"""
Run Supabase migrations using Python client
"""

import os
from pathlib import Path
from supabase import create_client, Client

# Configuration
SUPABASE_URL = "https://xhohhxoowqlldbdcpynj.supabase.co"
SUPABASE_SERVICE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Inhob2hoeG9vd3FsbGRiZGNweW5qIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc1OTU5OTY4OSwiZXhwIjoyMDc1MTc1Njg5fQ.SNp0GFvhTQ-5RlZs8ZFpSrusiUFGxROe1AoOfPSVTok"

# Initialize Supabase client
supabase: Client = create_client(SUPABASE_URL, SUPABASE_SERVICE_KEY)

# Migrations directory
migrations_dir = Path(__file__).parent.parent / "supabase" / "migrations"

# Migrations to run (skip already completed ones)
migrations = [
    "003_storage_buckets.sql",
    "004_analytics_views.sql",
    "20250104_oauth_tokens.sql",
    "20250104_speakers_tables.sql",
]

def run_migration(filename: str) -> bool:
    """Run a single migration file"""
    filepath = migrations_dir / filename

    if not filepath.exists():
        print(f"   ⏭️  {filename} - File not found, skipping")
        return True

    try:
        with open(filepath, 'r') as f:
            sql = f.read()

        # Execute the SQL
        result = supabase.rpc('exec_sql', {'query': sql}).execute()

        print(f"   ✅ {filename}")
        return True

    except Exception as e:
        print(f"   ❌ {filename} - Error: {str(e)}")
        return False

def main():
    print("🚀 Running database migrations...\n")
    print(f"   Project: xhohhxoowqlldbdcpynj")
    print(f"   Total migrations: {len(migrations)}\n")

    for migration in migrations:
        success = run_migration(migration)
        if not success:
            print(f"\n❌ Migration failed: {migration}")
            print("Stopping migration process.")
            return 1

    print("\n✅ All migrations completed successfully!")
    print("\n📋 Next steps:")
    print("   1. Verify tables in Supabase Dashboard")
    print("   2. Deploy Edge Functions")
    print("   3. Test the application locally: npm run dev")
    return 0

if __name__ == "__main__":
    exit(main())

File: scripts/run-migrations.js
#!/usr/bin/env node

/**
 * Run all Supabase migrations
 * Executes SQL migration files in order
 */

const fs = require('fs');
const path = require('path');
const https = require('https');

const config = {
  projectRef: 'xhohhxoowqlldbdcpynj',
  serviceRoleKey: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Inhob2hoeG9vd3FsbGRiZGNweW5qIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc1OTU5OTY4OSwiZXhwIjoyMDc1MTc1Njg5fQ.SNp0GFvhTQ-5RlZs8ZFpSrusiUFGxROe1AoOfPSVTok',
  accessToken: 'sbp_7fe234f01d83baf98e33439358447099d8bf7fc4',
};

const migrationsDir = path.join(__dirname, '..', 'supabase', 'migrations');

const migrations = [
  // '001_initial_schema.sql',  // Already completed
  // '002_row_level_security.sql',  // Already completed
  // '003_storage_buckets.sql',  // Handled by setup-storage-buckets.js
  '004_analytics_views.sql',
  '20250104_oauth_tokens.sql',
  '20250104_speakers_tables.sql',
];

async function executeSql(sql, migrationName) {
  return new Promise((resolve, reject) => {
    const data = JSON.stringify({ query: sql });

    const options = {
      hostname: 'api.supabase.com',
      port: 443,
      path: `/v1/projects/${config.projectRef}/database/query`,
      method: 'POST',
      headers: {
        'Authorization': `Bearer ${config.accessToken}`,
        'Content-Type': 'application/json',
        'Content-Length': data.length,
      },
    };

    const req = https.request(options, (res) => {
      let responseData = '';

      res.on('data', (chunk) => {
        responseData += chunk;
      });

      res.on('end', () => {
        if (res.statusCode === 200 || res.statusCode === 201) {
          console.log(`   ✅ ${migrationName}`);
          resolve();
        } else {
          console.error(`   ❌ ${migrationName} - Status: ${res.statusCode}`);
          try {
            const error = JSON.parse(responseData);
            console.error(`   Error: ${error.message || JSON.stringify(error)}`);
          } catch {
            console.error(`   Response: ${responseData}`);
          }
          reject(new Error(`Migration ${migrationName} failed`));
        }
      });
    });

    req.on('error', (error) => {
      console.error(`   ❌ ${migrationName} - Network error:`, error.message);
      reject(error);
    });

    req.write(data);
    req.end();
  });
}

async function runMigrations() {
  console.log('🚀 Running database migrations...\n');
  console.log(`   Project: ${config.projectRef}`);
  console.log(`   Total migrations: ${migrations.length}\n`);

  for (const migration of migrations) {
    const migrationPath = path.join(migrationsDir, migration);

    if (!fs.existsSync(migrationPath)) {
      console.log(`   ⏭️  ${migration} - File not found, skipping`);
      continue;
    }

    try {
      const sql = fs.readFileSync(migrationPath, 'utf-8');
      await executeSql(sql, migration);
    } catch (error) {
      console.error(`\n❌ Migration failed: ${migration}`);
      console.error(`   Error: ${error.message}`);
      console.log('\nStopping migration process.');
      process.exit(1);
    }
  }

  console.log('\n✅ All migrations completed successfully!');
  console.log('\n📋 Next steps:');
  console.log('   1. Verify tables in Supabase Dashboard');
  console.log('   2. Deploy Edge Functions');
  console.log('   3. Test the application locally: npm run dev');
}

runMigrations().catch(error => {
  console.error('\n❌ Migration process failed:', error.message);
  process.exit(1);
});
